embedded/sql/stmt.go

```diff
@@ -591,9 +591,11 @@ func (stmt *UpsertIntoStmt) compileUsing(e *Engine, implicitDB *Database, params
 
 		constraint := store.NoConstraint
 
-		if stmt.isInsert {
+		if stmt.isInsert && !table.pk.autoIncrement {
```

Comment:
:+1: Nice catch! I didn't realize before that Upsert is also used for standard Insert.

Out of curiosity - why not the other way around so that the Upsert is a special case of Insert?

Reply:
Upserts comes easier than Insert. Insert requires constraint checking while Upsert is able to work without any checking. That's why Insert functionality comes afterwards